tensorflow-tools/src/main/java/org/tensorflow/tools/buffer/impl/adapter/DoubleDataBufferAdapter.java

```diff
@@ -56,16 +56,48 @@ public DoubleDataBuffer write(double[] src, int offset, int length) {
     return this;
   }
 
+  @Override
+  public DoubleDataBuffer copyTo(DataBuffer<Double> dst, long size) {
+    Validator.copyToArgs(this, dst, size);
+    if (dst instanceof DoubleDataBuffer) {
+      for (long idx = 0L; idx < size; ++idx) {
+        ((DoubleDataBuffer)dst).setDouble(getDouble(idx), idx);
```

Comment:
Might be best to hoist the cast out of the for loop in case the JIT doesn't figure it out. Pity we can't use the pattern matching from JDK 14 yet.